sql/src/stream.cc

```diff
@@ -984,19 +987,32 @@ void stream::_process_host_check(
     }
 
     // Processing.
-    _host_check_update << hc;
-    try {
-      _host_check_update.run_statement();
+    unsigned int str_hash(qHash(hc.command_line));
+    // Did the command changed since last time?
+    if (_cache_hst_cmd[hc.host_id] != str_hash) {
```

Comment:
As the cache entry might not exist yet, you should check for the entry existence first and compare string hashes only if the entry exist.

Reply:
operator[] creates it if it does not exist. The new value will be 0 in that case and it will be changed because the condition will be true.
I suggest a little improvement, that is
`unsigned int& cmd_hash(_cache_hst_cmd[hc.host_id]);
if (cmd_hash != str_hash) {
...
cmd_hash = str_hash;
`
That way, the operator is called only one time. :-)

Reply:
> The new value will be 0 in that case and it will be changed because the condition will be true.

It might not be changed. The return value of qHash() could be 0.